Lib/doctest.py

```diff
@@ -1393,7 +1393,13 @@ def __run(self, test, compileflags, out):
 
             # The example raised an exception:  check if it was expected.
             else:
-                exc_msg = traceback.format_exception_only(*exception[:2])[-1]
+                formatted_ex = traceback.format_exception_only(*exception[:2])
+                if issubclass(exception[0], SyntaxError):
+                    # SyntaxError / IndentationError is special:
+                    # we don't care about the carets / suggestions / etc
+                    exc_msg = formatted_ex[-1]
```

Comment:
Is this correct? What about the notes? I think we need a test for SyntaxError because it's different.

Reply:
Indeed, it is not. SyntaxError with notes does not pass. Fixing!